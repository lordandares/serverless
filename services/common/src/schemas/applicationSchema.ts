import { InferType, mixed, object, string } from 'yup'

export const applicationSchema = object().shape({
  _id: string(),
  authenticationStrategies: mixed(),
  flags: mixed(),
  name: string().required(),
  plugins: mixed(),
  settings: mixed(),
  speakerbox: object().shape({
    consumerId: string(),
    token: string(),
  }),
  theme: object().shape({
    logos: object(),
  }),
})

export type ApplicationSchema = InferType<typeof applicationSchema>

export default applicationSchema
