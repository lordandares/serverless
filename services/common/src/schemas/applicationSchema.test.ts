import { omit } from 'lodash/fp'
import { ValidationError } from 'yup'
import { applicationSchema } from './applicationSchema'

const validApplication = {
  _id: '5b480b46fbef0ba17da8f426',
  name: 'Default App 1',
  settings: {
    duressTypes: ['Fire', 'Intruder', 'Injury'],
    gps: {
      accuracy: 10,
    },
    location: {
      passive: {
        enabled: true,
        options: {
          distanceFilter: 50,
          preventSuspend: true,
        },
      },
      adhoc: {
        enabled: true,
        options: {
          timeout: 20,
        },
      },
    },
    uuids: [
      'AAAAAAAA-AAAA-4AAA-AAAA-AAAAAAAAAAAA',
      'C7DC932D-A1A7-4DCB-931A-8FCA647A7E01',
    ],
  },
  plugins: {
    loops: {
      enabled: true,
    },
    serviceNow: {
      enabled: true,
    },
    slack: {
      enabled: false,
    },
    winteam: {
      enabled: true,
      options: {
        configSetting: 1,
        anotherConfigSetting: 2,
      },
      secrets: {
        awsSecretId: 'lio/serverless/winteam-service/test/default-app-1',
      },
    },
  },
  flags: {
    debug: true,
  },
  theme: {
    logos: {
      pdf: 'https://s3.amazonaws.com/assets-lighthouse-io/img/logo-medium.png',
    },
  },
  timezone: 'Australia/Melbourne',
}

test('valid', () => {
  expect.assertions(1)

  const validate = applicationSchema.validate(validApplication, {
    strict: true,
  })

  return expect(validate).resolves.toMatchSnapshot()
})

test('missing name', () => {
  expect.assertions(1)

  const validate = applicationSchema.validate(omit('name', validApplication), {
    strict: true,
  })

  return expect(validate).rejects.toThrowError(ValidationError)
})
