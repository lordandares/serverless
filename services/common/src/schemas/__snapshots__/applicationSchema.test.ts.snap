// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`valid 1`] = `
Object {
  "_id": "5b480b46fbef0ba17da8f426",
  "flags": Object {
    "debug": true,
  },
  "name": "Default App 1",
  "plugins": Object {
    "loops": Object {
      "enabled": true,
    },
    "serviceNow": Object {
      "enabled": true,
    },
    "slack": Object {
      "enabled": false,
    },
    "winteam": Object {
      "enabled": true,
      "options": Object {
        "anotherConfigSetting": 2,
        "configSetting": 1,
      },
      "secrets": Object {
        "awsSecretId": "lio/serverless/winteam-service/test/default-app-1",
      },
    },
  },
  "settings": Object {
    "duressTypes": Array [
      "Fire",
      "Intruder",
      "Injury",
    ],
    "gps": Object {
      "accuracy": 10,
    },
    "location": Object {
      "adhoc": Object {
        "enabled": true,
        "options": Object {
          "timeout": 20,
        },
      },
      "passive": Object {
        "enabled": true,
        "options": Object {
          "distanceFilter": 50,
          "preventSuspend": true,
        },
      },
    },
    "uuids": Array [
      "AAAAAAAA-AAAA-4AAA-AAAA-AAAAAAAAAAAA",
      "C7DC932D-A1A7-4DCB-931A-8FCA647A7E01",
    ],
  },
  "theme": Object {
    "logos": Object {
      "pdf": "https://s3.amazonaws.com/assets-lighthouse-io/img/logo-medium.png",
    },
  },
  "timezone": "Australia/Melbourne",
}
`;
